contracts/periphery/mintburn/HyperCoreFlowExecutor.sol

```diff
@@ -591,23 +604,35 @@ contract HyperCoreFlowExecutor is AccessControlUpgradeable, AuthorizedFundedFlow
     function _initiateSwapFlow(CommonFlowParams memory params, uint256 maxUserSlippageBps) internal {
         address initialToken = baseToken;
 
-        // Check account activation
-        if (!HyperCoreLib.coreUserExists(params.finalRecipient)) {
-            if (params.maxBpsToSponsor > 0) {
-                revert AccountNotActivatedError(params.finalRecipient);
-            } else {
-                emit AccountNotActivated(params.quoteNonce, params.finalRecipient);
-                params.finalToken = initialToken;
-                _fallbackHyperEVMFlow(params);
-                return;
-            }
-        }
-
         MainStorage storage $ = _getMainStorage();
         CoreTokenInfo memory initialCoreTokenInfo = $.coreTokenInfos[initialToken];
         CoreTokenInfo memory finalCoreTokenInfo = $.coreTokenInfos[params.finalToken];
         FinalTokenInfo memory finalTokenInfo = _getExistingFinalTokenInfo(params.finalToken);
 
+        bool userAccountExists = HyperCoreLib.coreUserExists(params.finalRecipient);
+        if (!userAccountExists) {
+            if (params.accountCreationMode == AccountCreationMode.Standard) {
+                if (params.maxBpsToSponsor > 0) {
+                    revert AccountNotActivatedError(params.finalRecipient);
+                } else {
+                    emit AccountNotActivated(params.quoteNonce, params.finalRecipient);
+                    params.finalToken = initialToken;
+                    _fallbackHyperEVMFlow(params);
+                    return;
+                }
+            } else {
+                // In the FromUserFunds logic for the swap flow, we only allow account creation if the final token is
+                // usable for it. Finilazing the swap flow should handle account creation automatically when doing the
+                // final transfer of tokens to the user
+                if (!finalCoreTokenInfo.canBeUsedForAccountActivation) {
+                    emit AccountNotActivated(params.quoteNonce, params.finalRecipient);
+                    params.finalToken = initialToken;
+                    _fallbackHyperEVMFlow(params);
+                    return;
+                }
+            }
+        }
```

Comment:
also this could be simplified to something like:

```
if (!HyperCoreLib.coreUserExists(params.finalRecipient)) {
    bool isStandard = params.accountCreationMode == AccountCreationMode.Standard;
    
    if (isStandard && params.maxBpsToSponsor > 0) {
        revert AccountNotActivatedError(params.finalRecipient);
    }
    
    // Fall back unless FromUserFunds mode with an eligible activation token
    if (isStandard || !finalCoreTokenInfo.canBeUsedForAccountActivation) {
        emit AccountNotActivated(params.quoteNonce, params.finalRecipient);
        params.finalToken = initialToken;
        _fallbackHyperEVMFlow(params);
        return;
    }
}
```